File: frontend/src/types/ingest.ts
export interface IngestSource {
  id: string;
  source_type: "file" | "sheet" | "api_push" | "api_pull" | "manual";
  name: string;
  config: Record<string, unknown>;
  secret_ref?: string | null;
  target_entity?: string | null;
  is_active: boolean;
  created_at?: string | null;
}

export interface IngestJob {
  id: string;
  source_id?: string | null;
  trigger: string;
  status: "running" | "succeeded" | "failed" | "partial";
  total: number;
  imported: number;
  skipped: number;
  failed: number;
  pending_review: number;
  error_summary?: string | null;
  created_at?: string | null;
}

export interface IngestItem {
  id: string;
  job_id: string;
  target_entity: string;
  status: "pending" | "imported" | "skipped" | "failed" | "needs_review";
  confidence: "high" | "medium" | "low";
  source_locator?: string | null;
  raw_payload: Record<string, unknown>;
  error?: string | null;
  review_note?: string | null;
  /** 后端给的建议默认勾选状态：仅 high/medium 且 pending 时为 true */
  default_checked: boolean;
}

export interface ConfirmResult {
  confirmed: number;
  failed: Array<{ item_id: string; reason: string }>;
}

export interface SkipResult {
  skipped: number;
}

File: backend/app/schemas/ingest.py
"""DataHub 出入参。"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class SourceIn(BaseModel):
    source_type: str = Field(pattern="^(file|sheet|api_push|api_pull|manual)$")
    name: str = Field(min_length=1, max_length=200)
    config: dict = Field(default_factory=dict)
    secret_ref: Optional[str] = Field(default=None, max_length=120)
    target_entity: Optional[str] = Field(default=None, max_length=60)
    is_active: bool = True


class SourceOut(SourceIn):
    id: str
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class JobOut(BaseModel):
    id: str
    source_id: Optional[str] = None
    trigger: str
    status: str
    total: int
    imported: int
    skipped: int
    failed: int
    pending_review: int
    error_summary: Optional[str] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class ItemOut(BaseModel):
    id: str
    job_id: str
    target_entity: str
    status: str
    confidence: str
    source_locator: Optional[str] = None
    raw_payload: dict
    error: Optional[str] = None
    """前端用于决定复选框默认状态：仅 high/medium 默认勾选。"""
    default_checked: bool = True


class ConfirmIn(BaseModel):
    item_ids: list[str] = Field(min_length=1)
